ocaml/testsuite/tests/typing-local/local.ml

```diff
@@ -2767,6 +2767,15 @@ Error: This function or one of its parameters escape their region
        when it is partially applied.
 |}];;
 
+let f () = ((fun x -> function | 0 -> x | y -> x + y) : (local_ int -> (int -> int)));;
```

Comment:
I think I expect this to be accepted, because of mode-crossing. Maybe we can't mode-cross in the middle of an n-ary function? But I don't see why.

Reply:
Good point - when typing a function, the `arg_mode` is not mode crossed. I can fix that in a separate PR. Meanwhile, for the tests here I will use `string`. 

Speaking of this, it's really easy to forget spots to cross modes. Maybe we should make `expected_mode.submode` takes a `type_expr` and handle mode crossing in a single spot.